tests/contracts/nonfungible-token/contract.scilla

```diff
@@ -0,0 +1,411 @@
+(* Checks if a given address is approved to make txn the given tokenID *)
+(* Not to be confused with isApprovedForAll                            *)
+let isApproved = 
+    fun (msgSender: Address) =>
+    fun (tokenID: Uint256) =>
+    fun (approvalMap_tmp: Map (Uint256) (Address)) =>
+        let val = builtin get approvalMap_tmp tokenID in
+        match val with
+        | None => False 
+        | Some val =>
+            let isSenderApproved = builtin eq val msgSender in
+            match isSenderApproved with
+            | True => True 
+            | False => False 
+            end
+        end
+
+(* Checks if an message sender is approved by a given owner. (i.e. operator) *)
+let isApprovedForAll = 
+    fun (msgSender: Address) =>
+    fun (tokenOwner: Address) =>
+    fun (operatorMap: Map (Address) (Map (Address) (Bool))) =>
+        let m = builtin get operatorMap tokenOwner in
+        match m with
+        | None => False
+            (* owner did not assign anyone to the approval mapping *)
+        | Some val =>
+            (* val is of type Map (Address) (Bool) *)
+            let check_list = builtin get val msgSender in
+            match check_list with
+            | None => False
+            | Some is_sender_approved =>
+                (* check if sender has access rights *)
+                match is_sender_approved with
+                | True => True
+                | False => False
+                end
+            end
+        end
+
+(* Check if a sender is an operator of the owner, approved for the given ID *)
+(* or is the owner of the token *)
+let isApprovedOrOwner =
+    fun (isOwner: Bool) =>
+    fun (isApproved: Bool) =>
+    fun (isApprovedForAll: Bool) =>
+        let checkone =  orb isOwner isApproved in
+        let checktwo = orb checkone isApprovedForAll in
+        match checktwo with
+        | True => True
+        | False => False
+        end
+
+(* Error codes *)
+let code_success = Uint32 0
+let code_failure = Uint32 1
+let code_not_authorized = Uint32 2
+let code_not_found = Uint32 4
+let code_bad_request = Uint32 5
+let code_token_exists = Uint32 6
+let code_unexpected_error = Uint32 9
+
+
+(***************************************************)
+(*             The contract definition             *)
+(***************************************************)
+
+contract NonfungibleToken
+(contractOwner : Address,
+ name : String,
+ symbol: String
+)
+
+(* mutable fields *)
+
+(* Mapping between tokenId to token owner *)
+field tokenOwnerMap: Map Uint256 Address = Emp Uint256 Address
+(* Mapping from owner to number of owned tokens *)
+field ownedTokenCount: Map Address Uint256 = Emp Address Uint256
+
+(* Mapping between tokenId to approved address *)
+(* @dev: There can only be one approved address per token at a given time. *)
+field tokenApprovals: Map Uint256 Address = Emp Uint256 Address
+
+(* Mapping from owner to operator approvals  *)
+field operatorApprovals: Map Address Map Address Bool 
```

Comment:
Minor: Would `Map Address (Map Address Bool)` improve readability?

Reply:
Fixed in 56d20bc015ad596d19f8f7af9c25bb7b4d1a472d